jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/CompactPathRule.java

```diff
@@ -35,7 +35,7 @@ public CompactPathRule()
     @Override
     public void applyURI(Request request, String oldURI, String newURI) throws IOException
     {
-        String uri = request.getRequestURI();
+        String uri = oldURI;
```

Comment:
This is the key change, don't use the servlet `getRequestURI()` as that doesn't include the query section, but instead use the pass-in `oldURI` which is the path+query.